### PR TITLE
Memoization rollback with previous PR feedback incorporated

### PR DESCRIPTION
**Pull request by karlbaker02**

### CODE REVIEW

File: src/main/java/uk/gov/register/core/TransactionalMemoizationStore.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        HashMap<Integer, byte[]> hashesBySize = stagedHashes.get(start);
+        hashesBySize.put(size, value);
+    }
+
+    @Override
+    public byte[] get(Integer start, Integer size) {
+        byte[] hash = null;
+
+        if (stagedHashes.containsKey(start)) {
+            hash = stagedHashes.get(start).get(size);
```
Comment (philandstuff) [inline src/main/java/uk/gov/register/core/TransactionalMemoizationStore.java]: why not just `return hash` here?

Comment (karlbaker02) [inline src/main/java/uk/gov/register/core/TransactionalMemoizationStore.java]: This is what I had meant to do, thanks for pointing out.

Comment (philandstuff) [inline src/main/java/uk/gov/register/core/TransactionalMemoizationStore.java]: in which case, I'd find `if (hash != null) { return hash; }` less confusing here than the ternary currently used below

File: src/main/java/uk/gov/register/core/TransactionalMemoizationStore.java [hunk 2]
```diff
@@ -14,30 +15,21 @@ public TransactionalMemoizationStore(MemoizationStore memoizationStore) {
...
+        stagedHashes.put(Pair.of(start, size), value);
     }
 
     @Override
     public byte[] get(Integer start, Integer size) {
-        byte[] hash = null;
-
-        if (stagedHashes.containsKey(start)) {
-            hash = stagedHashes.get(start).get(size);
+        if (stagedHashes.containsKey(Pair.of(start, size)))  {
```
Comment (chr1sdi) [inline src/main/java/uk/gov/register/core/TransactionalMemoizationStore.java]: maybe assigning `Pair.of(start, size)` to an variable would be clearer and we can reuse it in the `get`?
somink like: `Pair<Integer, Integer> startSizePair = Pair.of(start, size);`  ? what do you think?

Comment (chr1sdi) [inline src/main/java/uk/gov/register/core/TransactionalMemoizationStore.java]: if we assume that there is not going to be a `null` hash value for any star, size pair we can even compress it to a 1 liner, something like this:
```
public byte[] get(Integer start, Integer size) {
        return stagedHashes.getOrDefault(Pair.of(start, size), memoizationStore.get(start, size));
    }
```

**Comment by chr1sdi** [comment]
👍

**Pull request merged**